src/hash.h

```diff
@@ -165,6 +166,39 @@ class CHashWriter : public HashWriter
 };
 
 /** Reads data from an underlying stream, while hashing the read data. */
+template <typename Source>
+class HashVerifier : public HashWriter
+{
+private:
+    Source& m_source;
```

Comment:
Shouldn't this ideally be `const`? Can be done when making `AutoFile::read()` `const`.

<details>
<summary>git diff</summary>

```diff
diff --git a/src/hash.h b/src/hash.h
index b2ef29fcd..ef5f03578 100644
--- a/src/hash.h
+++ b/src/hash.h
@@ -170,10 +170,10 @@ template <typename Source>
 class HashVerifier : public HashWriter
 {
 private:
-    Source& m_source;
+    const Source& m_source;
 
 public:
-    explicit HashVerifier(Source& source LIFETIMEBOUND) : m_source{source} {}
+    explicit HashVerifier(const Source& source LIFETIMEBOUND) : m_source{source} {}
 
     void read(Span<std::byte> dst)
     {
diff --git a/src/streams.h b/src/streams.h
index 4f2c3ffe7..ff812280f 100644
--- a/src/streams.h
+++ b/src/streams.h
@@ -516,7 +516,7 @@ public:
     //
     // Stream subset
     //
-    void read(Span<std::byte> dst)
+    void read(Span<std::byte> dst) const
     {
         if (!file) throw std::ios_base::failure("AutoFile::read: file handle is nullptr");
         if (fread(dst.data(), 1, dst.size(), file) != dst.size()) {
```
</details>

Reply:
My preference would be to do it for all functions in the same commit: `git grep 'void read(Span<'`